net.py

```diff
@@ -1,172 +0,0 @@
-import numpy as np
-
-class Net:
-    def __init__(self, layers):
-        self.W = []
-        self.dW = []
-        self.b = []
-        self.db = []
-        self.A_saved = []
-        self.Z_saved = []
-        self.dA = []
-        self.dZ = []
-        self.backup = []
-        self.L = len(layers) - 1
-        self.layers = layers
-        # -1 are appended to arrays in order to assign n-th index to n-th layer, e.g. W2 = W[2]
-        self.W.append(-1)
-        self.b.append(-1)
-        self.A_saved.append(-1)
-        self.Z_saved.append(-1)
-        self.dA.append(-1)
-        self.dW.append(-1)
-        self.db.append(-1)
-        self.dZ.append(-1)
-        for i in range(1,self.L+1):
-            #print(np.sqrt(1/layers[i-1]))
-            self.W.append(np.random.randn(layers[i], layers[i-1]) * 0.01) #np.sqrt(2/layers[i-1]))
-            self.b.append(np.zeros((layers[i], 1)))
-            self.A_saved.append(-1)
-            self.Z_saved.append(-1)
-            self.dW.append(-1)
-            self.db.append(-1)
-            self.dA.append(-1)
-            self.dZ.append(-1)
-        #self.W[-1] = self.W[-1] * 0.5   # different activation function in last layer
-
-    def print_me(self):
-        print(self.dW)
-
-    @staticmethod
-    def activation(Z, type):
-        if type == 'relu':
-            return np.maximum(0,Z)
-        elif type == 'sigmoid':
-            return 1/(1+np.exp(-Z))
-        elif type == 'softmax':
-            temp = np.exp(Z)
-            return temp/np.sum(temp)
-        else:
-            print('something went wrong')
-            quit()
-
-    def forward_prop(self, X):
-        A = X
-        for i in range(1, self.L):
-            #print(i)
-            Z = np.dot(self.W[i], A) + self.b[i]
-            A = Net.activation(Z, 'relu')
-            self.A_saved[i] = A
-            self.Z_saved[i] = Z
-        # for last layer we use softmax or sigmoid accordingly to output layer size
-        Z = np.dot(self.W[-1], A) + self.b[-1]
-        if len(self.b[-1]) == 1:
-            A = Net.activation(Z, 'sigmoid')
```

Comment:
Instead 
```
        if len(self.b[-1]) == 1:
            A = Net.activation(Z, 'sigmoid')
```
maybe set mode? like
```
        mode = sigmoid if len(self.b[-1]) == 1 else 'softmax'
        A = Net.activation(Z, mode)

```
In this situation logic is in one place ( if you want to add new parameter to activation, you need to edit only one line). Try to control logic by variables/parameters. It's more readable and generic.